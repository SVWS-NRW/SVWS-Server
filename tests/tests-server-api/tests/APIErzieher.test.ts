import { describe, test, expect } from 'vitest'
import { getApiServer } from "./utils/TestUtils.js";
import { Erzieherart, ErzieherListeEintrag } from "@core";

describe("Erzieher Tests", () => {
	describe.each([{schema: "GymAbi01"}])('gegen %s', ({schema}) => {
		const api = getApiServer(schema);

		test('GET: Ein ErzieherStammdaten Objekt', async () => {
			const result = await api.getErzieherStammdaten(schema, 19941);
			expect(result).toMatchSnapshot();
		})

		test('GET: Ein ErzieherStammdaten Objekt, das nicht existiert', async () => {
			await expect(api.getErzieherStammdaten(schema, 99999)).rejects.toThrowError(`Fetch failed for GET: /db/${schema}/erzieher/99999/stammdaten`);
		})

		// TODO: Fix me
		test.skip("getErzieher", async () => {
			const result = await api.getErzieher(schema);
			expect(result).toMatchSnapshot();
			expect(result.toArray()[0]).toBeInstanceOf(ErzieherListeEintrag);
		});

		test("getErzieherArten", async () => {
			const result = await api.getErzieherArten(schema);
			expect(result).toMatchSnapshot();
			expect(result.toArray()[0]).toBeInstanceOf(Erzieherart);
		});
	})
});
