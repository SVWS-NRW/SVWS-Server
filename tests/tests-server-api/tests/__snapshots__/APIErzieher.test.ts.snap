// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Erzieher Tests > gegen { schema: 'GymAbi01' } > GET: Ein ErzieherStammdaten Objekt 1`] = `
ErzieherStammdaten {
  "anrede": "Frau",
  "bemerkungen": null,
  "eMail": "Blum@email.de",
  "erhaeltAnschreiben": true,
  "hausnummer": "443",
  "hausnummerZusatz": "",
  "id": 19941,
  "idErzieherArt": 5,
  "idSchueler": 1001,
  "nachname": "Blum",
  "ortsteilID": null,
  "staatsangehoerigkeitID": null,
  "strassenname": "Militscher Weg",
  "titel": null,
  "vorname": "Bianca",
  "wohnortID": 698,
}
`;

exports[`Erzieher Tests > gegen { schema: 'GymAbi01' } > getErzieherArten 1`] = `
ArrayList {
  "addFirst": [Function],
  "addLast": [Function],
  "elementData": [
    Erzieherart {
      "bezeichnung": "Vater",
      "exportBez": "",
      "id": 1,
      "istSichtbar": true,
      "sortierung": 1,
    },
    Erzieherart {
      "bezeichnung": "Mutter",
      "exportBez": "",
      "id": 2,
      "istSichtbar": true,
      "sortierung": 2,
    },
    Erzieherart {
      "bezeichnung": "Schüler ist volljährig",
      "exportBez": "",
      "id": 3,
      "istSichtbar": true,
      "sortierung": 3,
    },
    Erzieherart {
      "bezeichnung": "Schülerin ist volljährig",
      "exportBez": "",
      "id": 4,
      "istSichtbar": true,
      "sortierung": 4,
    },
    Erzieherart {
      "bezeichnung": "Eltern",
      "exportBez": "",
      "id": 5,
      "istSichtbar": true,
      "sortierung": 5,
    },
    Erzieherart {
      "bezeichnung": "Sonstige",
      "exportBez": "",
      "id": 6,
      "istSichtbar": true,
      "sortierung": 6,
    },
    Erzieherart {
      "bezeichnung": "Partner",
      "exportBez": "",
      "id": 7,
      "istSichtbar": true,
      "sortierung": 7,
    },
    Erzieherart {
      "bezeichnung": "Gastmutter",
      "exportBez": "",
      "id": 8,
      "istSichtbar": true,
      "sortierung": 8,
    },
    Erzieherart {
      "bezeichnung": "Großeltern",
      "exportBez": "",
      "id": 9,
      "istSichtbar": true,
      "sortierung": 9,
    },
    Erzieherart {
      "bezeichnung": "Pflegeeltern (Vollzeitpflege)",
      "exportBez": "",
      "id": 10,
      "istSichtbar": true,
      "sortierung": 10,
    },
    Erzieherart {
      "bezeichnung": "Kinderheim",
      "exportBez": "",
      "id": 11,
      "istSichtbar": true,
      "sortierung": 11,
    },
  ],
  "getFirst": [Function],
  "getLast": [Function],
  "modCount": 11,
  "removeFirst": [Function],
  "removeLast": [Function],
}
`;
